### PR TITLE
Add support for the path endpoints

### PR DESCRIPTION
**Pull request by astrinski-smartsheet**
# Pull Request

## Description
Added support and WireMock tests for the following endpoints:
- /2.0/folders/{folderId}/path
- /2.0/reports/{reportId}/path
- /2.0/sheets/{sheetId}/path
- /2.0/sights/{sightId}/path

## Type of Change
- [ ] Bug fix
- [x] New feature
- [ ] Documentation update
- [ ] Code refactoring
- [ ] Other (please describe):

## Environment Information
- Smartsheet API Version: [e.g. 2.0]
- Smartsheet Python SDK Version: [e.g. 3.0.0]
- Python Version: [e.g. 3.10]

## Testing

Manual testing with a simple test project

## Checklist

- [ ] My code follows the [style guidelines](../../../CONTRIBUTING.md#code-style-and-quality) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have updated the relevant files in `docs-source/` (see [Documentation guidelines](../../../CONTRIBUTING.md#documentation))
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: smartsheet/models/folder_path_node.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        """Return a Unix-style path string of folder names from this node to the target folder.
+
+        Unlike get_sheet_path/get_report_path/get_sight_path, this method returns the
+        deepest folder's name even when the root node has no name (e.g. a workspace root),
+        because the target IS a folder rather than an asset nested inside one. An empty
+        FolderPathNode (no id, no name, no children) returns an empty string.
+        """
+        if self.folders:
+            child_path = self.folders[0].get_folder_path()
+            return f"{self.name}/{child_path}" if self.name and child_path else self.name or child_path or ""
```
Comment (ggoranov-smar) [inline smartsheet/models/folder_path_node.py]: Add the leading forward slash to make it more "unix-like".
The same applies for the rest of the `get_{{asset}}_path` methods.

File: smartsheet/models/folder_path_node.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        self._folders.load(value)
+
+    def get_folder(self):
+        """Return the deepest FolderPathNode reachable from this node."""
+        if self.folders:
+            return self.folders[0].get_folder()
+        return self
+
+    def get_folder_path(self):
+        """Return a Unix-style path string of folder names from this node to the target folder.
```
Comment (ggoranov-smar) [inline smartsheet/models/folder_path_node.py:53]: An example of how the path looks would be great.
The same applies for the rest of the `get_{{asset}}_path` methods.

File: smartsheet/models/path_node.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        self._permalink = String()
+
+        if props:
+            deserialize(self, props)
+
+        self.request_response = None
+        self.__initialized = True
+
+    def __getattr__(self, key):
+        if key == "id":
```
Comment (ggoranov-smar) [inline smartsheet/models/path_node.py]: What is the goal of `self._id_` when this manages the `.id` prop?

File: smartsheet/models/folder_path_node.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+    @property
+    def folders(self):
+        return self._folders
+
+    @folders.setter
+    def folders(self, value):
+        self._folders.load(value)
+
+    def get_folder(self):
```
Comment (ggoranov-smar) [inline smartsheet/models/folder_path_node.py]: Consider naming it `get_leaf_folder'.
The same applies for the other similar methods.

File: smartsheet/folders.py [hunk 5]
```diff
@@ -245,6 +245,26 @@ def get_folder_metadata(self, folder_id, include=None) -> Union[Folder, Error]:
 
         return response
 
+    def get_folder_path(self, folder_id) -> Union[FolderPathNode, Error]:
```
Comment (ggoranov-smar) [inline smartsheet/folders.py]:
```suggestion
    def get_folder_path(self, folder_id: int) -> Union[FolderPathNode, Error]:
```

The same applies for the rest of the endpoint methods.

**Review by ggoranov-smar — CHANGES_REQUESTED** [review]

**Review by ggoranov-smar — CHANGES_REQUESTED** [review]
## Review

Verified against OpenAPI spec and WireMock mappings. All previous review comments have been addressed:
- ✅ Forward slashes in path strings
- ✅ `get_leaf_*` naming convention applied
- ✅ Type hints `Union[..., Error]` on all 4 methods
- ✅ `self._id` (no trailing underscore) in `PathNode`
- ✅ WireMock scenario names match mappings exactly
- ✅ Test assertion patterns use whole-object `to_dict() ==` comparisons
- ✅ All required test types present (nested + root-level variants + 4xx + 5xx)

Two issues remain:

### 1. Missing example in `FolderPathNode.get_leaf_folder_path()`

The review comment asked for examples in all `get_<asset>_path` methods. `SheetPathNode`, `ReportPathNode`, and `SightPathNode` all have `Example: '/Workspace/Folder/Sheet'` in their docstrings, but `FolderPathNode.get_leaf_folder_path()` does not.

### 2. `get_leaf_folder_path()` returns `'/None'` on empty node

When called on a `FolderPathNode()` with no `name` set, `get_leaf_folder_path()` returns `'/None'` instead of `None`. The test comment at line 97 says "empty node returns `/Root`" but that test constructs a node with `name="Root"` — a truly empty node (no props) produces `'/None'`.

The other three models return `None` from `get_leaf_*_path()` when there is nothing to traverse, which is the correct sentinel. `get_leaf_folder_path()` should guard against a `None` name:

```python
def get_leaf_folder_path(self):
    """Return a Unix-style path string...
    Example: '/Workspace/Folder/TargetFolder'
    """
    if not self.name:
        return None
    if self.folders:
        return f"/{self.name}{self.folders[0].get_leaf_folder_path()}"
    return f"/{self.name}"
```

Also add a unit test for `FolderPathNode()` (no props) returning `None`, consistent with the `test_empty_node_returns_none` tests in the other three model test classes.

**Review by ggoranov-smar — APPROVED** [review]
## Review

Both remaining issues from the previous review are resolved:

- ✅ `get_leaf_folder_path()` docstring now includes `Example: '/Workspace/Folder/TargetFolder'`
- ✅ `get_leaf_folder_path()` guards against `None` name and returns `None` on empty node, with a corresponding `test_empty_node_returns_none` test

All previously verified items remain correct. Approved.

File: CHANGELOG.md [hunk 6]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [x.x.x] - Unreleased
 
+### Added
+
+- Added `get_sheet_path`, `get_report_path`, `get_sight_path`, and `get_folder_path` methods for retrieving the hierarchical folder/workspace path of an asset. Returns typed path node models (`SheetPathNode`, `ReportPathNode`, `SightPathNode`, `FolderPathNode`) with helper methods `get_<asset>()` and `get_<asset>_path()` for convenient traversal.
```
Comment (ggoranov-smar) [inline CHANGELOG.md]: Final nit, can you list each supported endpoint on a new line with the sdk method that supports it. See https://github.com/smartsheet/smartsheet-javascript-sdk/pull/187/changes#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed for an example.

Apply this to the rest of the SDKs.

**Review by ggoranov-smar — DISMISSED** [review]

**Review by ggoranov-smar — APPROVED** [review]
